src/components/form/RadioButton.tsx

```diff
@@ -1,49 +1,98 @@
-import React, { useRef } from 'react';
+import React, { useRef, useState } from 'react';
 
 import { HelpText, Radio } from '@digdir/design-system-react';
 import type { RadioProps } from '@digdir/design-system-react';
 
 import classes from 'src/components/form/RadioButton.module.css';
+import { DeleteWarningPopover } from 'src/components/molecules/DeleteWarningPopover';
 import { getPlainTextFromNode } from 'src/utils/stringHelper';
 
 export interface IRadioButtonProps extends Omit<RadioProps, 'children'> {
   showAsCard?: boolean;
   label?: string;
   helpText?: React.ReactNode;
   hideLabel?: boolean;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  alertOnChange?: boolean;
+  alertText?: string;
+  confirmChangeText?: string;
 }
 
-export const RadioButton = ({ showAsCard = false, label, helpText, hideLabel, ...rest }: IRadioButtonProps) => {
+export const RadioButton = ({
+  showAsCard = false,
+  label,
+  helpText,
+  hideLabel,
+  onChange,
+  alertOnChange,
+  alertText,
+  confirmChangeText,
+  ...rest
+}: IRadioButtonProps) => {
   const inputRef = useRef<HTMLInputElement>(null);
   const Label = (
     <div className={`${hideLabel ? 'sr-only' : ''} ${classes.radioLabelContainer}`}>
       {label}
       {helpText ? <HelpText title={getPlainTextFromNode(helpText)}>{helpText}</HelpText> : null}
     </div>
   );
-  if (showAsCard) {
-    return (
-      /** This element is only clickable for visual
+  const [popoverOpen, setPopoverOpen] = useState(false);
+  const [tempEvent, setTempEvent] = useState<React.ChangeEvent<HTMLInputElement>>();
```

Comment:
Maybe it would be better to use a `useRef` instead of a `useState` here? Seeing as it is not needed to trigger any effects and is not used during render.